app/views/components/_related-navigation.html.erb

```diff
@@ -27,12 +28,30 @@
               <%= construct_section_heading(section_title) %>
             </h3>
           <% end %>
-          <nav role="navigation" class="app-c-related-navigation__nav-section" aria-labelledby="related-nav-<%= section_title %>">
+          <nav role="navigation" class="app-c-related-navigation__nav-section" aria-labelledby="related-nav-<%= section_title %>" data-module="toggle">
             <ul class="app-c-related-navigation_link-list" data-module="track-click">
-              <% links.each do |link| %>
-                <li class="app-c-related-navigation__link">
-                  <%= link_to link[:text], link[:path], class: "app-c-related-navigation__section-link" + construct_section_styling(section_title, defined_sections), rel: link[:rel] %>
-                </li>
+              <% constructed_link_array = [] %>
+              <% links.each.with_index(1) do |link, index| %>
+                <% if index <= max_section_length %>
+                  <li class="app-c-related-navigation__link">
+                    <%= link_to link[:text], link[:path], class: "app-c-related-navigation__section-link" + construct_section_styling(section_title, defined_sections), rel: link[:rel] %>
+                  </li>
+                <% else %>
+                  <% constructed_link_array.push(link_to link[:text], link[:path], class: "app-c-related-navigation__section-link" + construct_section_styling(section_title, defined_sections), rel: link[:rel]) %>
+                <% end %>
+              <% end %>
+              <% if links.length > max_section_length %>
+              <li class="app-c-related-navigation__link toggle-wrap">
+                <a href="#"
+                  data-controls="toggle_<%= section_title %>"
+                  data-expanded="false"
+                  data-toggled-text="<%= t("govuk_component.metadata.toggle_less", default: "Show fewer") %>">
+                    <%= t("govuk_component.metadata.toggle_more", number: remaining_links(links, max_section_length), default: "+ #{remaining_links(links, max_section_length)} more") %>
```

Comment:
This feels a bit weird, I wonder if we should fork this, I guess it works though! 🤔